modules/3d-tiles/src/tz3/tz3-archive.ts

```diff
@@ -0,0 +1,82 @@
+import md5 from 'md5';
+import {FileProvider, parseZipLocalFileHeader, HashElement, findBin} from '@loaders.gl/zip';
+import {DeflateCompression} from '@loaders.gl/compression';
+
+type CompressionHandler = (compressedFile: ArrayBuffer) => Promise<ArrayBuffer>;
+
+/**
+ * Handling different compression types in zip
+ */
+const COMPRESSION_METHODS: {[key: number]: CompressionHandler} = {
+  /** No compression */
+  0: async (compressedFile) => compressedFile,
+
+  /** Deflation */
+  8: async (compressedFile) => {
+    const compression = new DeflateCompression({raw: true});
+    const decompressedData = await compression.decompress(compressedFile);
+    return decompressedData;
+  }
+};
+
+/**
+ * Class for handling information about 3tz file
+ */
+export class TZ3Archive {
+  private tz3Archive: FileProvider;
+  private hashArray: HashElement[];
+  constructor(tz3Archive: FileProvider, hashFile: HashElement[]) {
+    this.tz3Archive = tz3Archive;
+    this.hashArray = hashFile;
+  }
+
+  /**
+   * Returns file with the given path from 3tz archive
+   * @param path - path inside the 3tz
+   * @returns buffer with ready to use file
+   */
+  async getFile(path: string): Promise<Buffer> {
```

Comment:
If at all possible, we do not want to expose Node.js buffer's in the loaders.gl API. They can technically be used in the browser but only because the bundler pulls in a big polyfill into the built code bundle, plus the applications now become more complex as they have to deal with both ArrayBuffer/Typed Arrays and Buffer objects for binary data.

Prefer ArrayBuffer if possible.